ruby-challenges.rb

```diff
@@ -10,18 +10,57 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+#create a method named includes_letter that uses .include? to check for the specific letters
+#also use .filter to return the wanted strings and leave out the unwanted strings in a new array
+
+   
+  def includes_letter(givenArray, string)
+    givenArray.filter {|newArray| newArray.include? string}
+  end
+
+  p includes_letter(beverages_array, letter_o)
+  p includes_letter(beverages_array, letter_t)
+
+  #output: ["coffee", "soda water"]
+  #output: ["tea", "water", "soda water"]
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+#from the hint, I found that .flatten takes all of the elements from multiple arrays and combines them into a new array, after that use .sort to get all of the states in alphabetical order but before all of that you have to use .values to access the values then you can apply .flatten and .sort
+
+p us_states.values.flatten.sort
```

Comment:
While this will provide the expected output, the prompt asks for a method as this will keep this code block dynamic.  The logic itself looks great, just wrapping it with a method will level this up.